eraser-dev/eraser
# Feat: Support Trivy status filtering

**What this PR does / why we need it**:
Adds ability to filter by image vulnerability status through a new field `ignoredStatuses` in `components.scanner.config`.
https://aquasecurity.github.io/trivy/v0.44/docs/configuration/filtering/#by-status

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #821 

**Special notes for your reviewer**:
Still needs to be (manually) end-to-end tested. Have only "verified" it works based on logic and the small test for the config, but have not done any manual verification.

## Manual Verification Steps
Install Eraser with updated scanner
1. `kind create cluster`
2. `make docker-build-trivy-scanner TRIVY_SCANNER_IMG=eraser-trivy-scanner:dev`
3. `kind load docker-image eraser-trivy-scanner:dev`
4. `helm upgrade --install eraser charts/eraser --values override.yaml` (seen below)

Test Eraser with vulnerable image
1. `docker pull docker.io/library/alpine:3.7.3` (pulling vulnerable image)
2. `docker exec -ti kind-control-plane bash`
3. `crictl images` (in the bash)
4. `kind load docker-image alpine:3.7.3`
    - Repeat this while updating the **override.yaml** and doing `helm upgrade --install eraser charts/eraser --values override.yaml` to see it taking effect. 
    - When `fixed` is an ignored status, the image does not get deleted. When we **don't** ignore it, it gets deleted (as expected). That alpine image only has a single `CRITICAL` - `fixed` vuln.

**override.yaml**
```yaml
runtimeConfig:
  manager:
    scheduling:
      repeatInterval: "1s"
  components:
    collector:
      enabled: true
      image:
        tag: "v1.2.1"
    scanner:
      enabled: true
      image:
        repo: "docker.io/library/eraser-trivy-scanner"
        tag: "dev"
      config: |
         cacheDir: /var/lib/trivy
         dbRepo: ghcr.io/aquasecurity/trivy-db
         deleteFailedImages: false
         deleteEOLImages: false
         vulnerabilities:
           ignoreUnfixed: false
           types:
             - os
             - library
           securityChecks:
             - vuln
           severities:
             - CRITICAL
           ignoredStatuses:
             - fixed
    remover:
      image:
        tag: "v1.2.1"

deploy:
  image:
    repo: ghcr.io/eraser-dev/eraser-manager
    pullPolicy: IfNotPresent
    tag: "v1.2.1"
```
